CMakePresets.json

```diff
@@ -0,0 +1,216 @@
+        "LDFLAGS": "-fuse-ld=lld"
+      }
+    },
+    {
+      "name": "nix-clang-debug",
+      "inherits": ["nix-base", "clang-base"],
+      "displayName": "Clang Debug Config",
+      "binaryDir": "${sourceDir}/build/clang/debug",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Debug",
+        "CMAKE_INSTALL_PREFIX": "${sourceDir}/install/clang/debug",
+        "VAST_ENABLE_ASAN": "ON",
+        "VAST_ENABLE_BUNDLED_CAF": "ON",
+        "VAST_PLUGINS": "${sourceDir}/plugins/pcap;${sourceDir}/plugins/sigma;${sourceDir}/plugins/summarize"
+      }
+    },
+    {
+      "name": "nix-clang-redeb",
+      "inherits": ["nix-base", "clang-base"],
+      "displayName": "Clang Redeb Config",
+      "binaryDir": "${sourceDir}/build/clang/redeb",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "RelWithDebInfo",
+        "CMAKE_INSTALL_PREFIX": "${sourceDir}/install/clang/redeb",
+        "VAST_ENABLE_BUNDLED_CAF": "OFF"
+      }
+    },
+    {
+      "name": "nix-clang-release",
+      "inherits": ["nix-base", "clang-base"],
+      "displayName": "Clang Release Config",
+      "binaryDir": "${sourceDir}/build/clang/release",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Release",
+        "CMAKE_INSTALL_PREFIX": "${sourceDir}/install/clang/release",
+        "VAST_ENABLE_BUNDLED_CAF": "OFF"
+      }
+    },
+    {
+      "name": "gcc-base",
+      "hidden": true,
+      "cacheVariables": {
+        "CMAKE_C_COMPILER": "gcc",
+        "CMAKE_CXX_COMPILER": "g++"
+      }
+    },
+    {
+      "name": "nix-gcc-debug",
+      "inherits": ["nix-base", "gcc-base"],
+      "displayName": "GCC Debug Config",
+      "binaryDir": "${sourceDir}/build/gcc/debug",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Debug",
+        "CMAKE_INSTALL_PREFIX": "${sourceDir}/install/gcc/debug",
+        "VAST_ENABLE_ASAN": "ON",
+        "VAST_ENABLE_BUNDLED_CAF": "ON",
+        "VAST_PLUGINS": "${sourceDir}/plugins/pcap;${sourceDir}/plugins/sigma;${sourceDir}/plugins/summarize"
+      },
+      "environment": {
+        "LDFLAGS": "-fuse-ld=lld"
+      }
+    },
+    {
+      "name": "nix-gcc-redeb",
+      "inherits": ["nix-base", "gcc-base"],
+      "displayName": "GCC Redeb Config",
+      "binaryDir": "${sourceDir}/build/gcc/redeb",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "RelWithDebInfo",
+        "CMAKE_INSTALL_PREFIX": "${sourceDir}/install/gcc/redeb",
+        "VAST_ENABLE_BUNDLED_CAF": "OFF"
+      },
+      "environment": {
+        "LDFLAGS": "-fuse-ld=lld"
+      }
+    },
+    {
+      "name": "nix-gcc-release",
+      "inherits": ["nix-base", "gcc-base"],
+      "displayName": "GCC Release Config",
+      "binaryDir": "${sourceDir}/build/gcc/release",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Release",
+        "CMAKE_INSTALL_PREFIX": "${sourceDir}/install/gcc/release",
+        "VAST_ENABLE_BUNDLED_CAF": "OFF"
+      },
+      "environment": {
+        "LDFLAGS": "-fuse-ld=gold"
```

Comment:
Perhaps `gold` too opinionated as a default?

Reply:
[Until mold is wrapped to be used with nixpkgs](https://github.com/NixOS/nixpkgs/pull/172452) we must choose between bfd and gold for linking LTO-enabled object files from GCC.
bfd is still the default, but I don't see a reason not to use gold on Linux.

Reply:
This build is not restricted to Linux, so that'd be one reason not to specify it.

Reply:
The only drawback of gold is that it can only produce ELF binaries, but the list of systems that support ELF is quite long:
https://en.wikipedia.org/wiki/Executable_and_Linkable_Format#Applications

gold is a very sensible default.